src/mixxxmainwindow.cpp

```diff
@@ -1159,14 +1160,14 @@ bool MixxxMainWindow::eventFilter(QObject* obj, QEvent* event) {
                         "DlgPreferences") {
             // return true for no tool tips
             switch (m_toolTipsCfg) {
-            case mixxx::TooltipsPreference::TOOLTIPS_ONLY_IN_LIBRARY:
+            case mixxx::preferences::constants::Tooltips::OnlyInLibrary:
```

Comment:
All the double :: look cluttered. Can we remove the "constants" namespace?

Reply:
we discussed that here https://github.com/mixxxdj/mixxx/pull/12798#issuecomment-1941300550

Reply:
Ah, I just read it. I can however not follow the reasoning. This PR introduces both "preferences::constants" there is no need to make it that long just to safe the "preferences" namespace for future use. We can introduce another independent namespace all the time. 

Btw: We have no policy of folder matching namespaces like in C#

Reply:
I have just checked how others do this: 
They define a namespace each type and than class less enums: 
https://github.com/zhuzichu520/FluentUI/blob/f830d5a9bfc0f6836e6e2aa7c80d7201b730013f/src/Def.h#L95
https://github.com/KDE/kdenlive/blob/7283b739900a01ef3caf86d458bedcf418ce1f60/src/definitions.h#L119

Lets follow them and not introduce  preferences and constants namespace. Just mixxx is sufficient IMHO.

Reply:
I don't like the class-less enums though. There is no reason to open up this enum to implicit conversions to `int` etc. wbout
```c++
namespace mixxx {
namespace Tooltips {
Q_NAMESPACE
enum class TooltipsOptions {...}
Q_ENUM_NS(TooltipsOptions)
using enum TooltipsOptions;
}
}
```
would that work for you?

Reply:
The reason is to get rid of the enum class namespace when using the enum in QML and use the extra namespace instead. 

But it also works for me if we get rid of at least one of the extra namespaces as you like.

Reply:
> The reason is to get rid of the enum class namespace when using the enum in QML and use the extra namespace instead.

Well, QML is a different discussion here. IIUC we can rename the type as we'd like. I'm primarily focused on the C++ ergonomics right now.